packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -0,0 +1,734 @@
+import * as codebuild from '@aws-cdk/aws-codebuild';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Contains information that defines how the build project reports the build status to the source provider.
+ */
+export interface BuildStatusConfig {
+  /**
+   * Specifies the context of the build status CodeBuild sends to the source provider.
+   */
+  readonly context: string;
+  /**
+   * Specifies the target url of the build status CodeBuild sends to the source provider.
+   */
+  readonly targetUrl: string;
+}
+
+/**
+ * Information about the build output artifacts for the build project.
+ */
+export interface ProjectArtifacts {
+  /**
+   * An identifier for this artifact definition.
+   *
+   * @default none
+   */
+  readonly artifactIdentifier?: string;
+  /**
+   *  Information that tells you if encryption for build artifacts is disabled.
+   *
+   * @default false
+   */
+  readonly encryptionDisabled?: boolean;
+  /**
+   * If this flag is set, a name specified in the buildspec file overrides the artifact name.
+   *
+   * @default false
+   */
+  readonly overrideArtifactName?: boolean;
+  /**
+   * The type of build output artifact.
+   */
+  readonly type: 'CODEPIPELINE' | 'S3' | 'NO_ARTIFACTS';
+  /**
+   * Information about the build output artifact location.
+   *
+   * @default none
+   */
+  readonly location?: String;
+  /**
+   * Along with namespaceType and name, the pattern that AWS CodeBuild will use to name and store the output artifact.
+   *
+   * @default none
+   */
+  readonly path?: String;
+  /**
+   * Along with path and name, the pattern that AWS CodeBuild will
+   * use to determine the name and location to store the output artifact.
+   *
+   * @default NONE
+   */
+  readonly namespaceType?: 'BUILD_ID' | 'NONE';
```

Comment:
the types of these inlined values should be in an interface with doc strings that explain what they are and help users in making the decision on what's appropriate for them